enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,322 @@
+        properties:
+          status:
+            properties:
+              platformStatus:
+                properties:
+                  azure:
+                    properties:
+                      resourceTags:
+                        description: resourceTags is a list of additional tags to apply to Azure
+                        resources created for the cluster. See 
+                        https://docs.microsoft.com/en-us/rest/api/resources/tags for information 
+                        on tagging Azure resources. Azure supports a maximum of 50 tags per 
+                        resource except for few, which have limitation of 15 tags. OpenShift 
+                        reserves 5 tags for its internal use, and allows 10 tags 
+                        for user configuration.
+                        type: array
+                        maxItems: 10
+                        items:
+                          description: AzureResourceTag is a tag to apply to Azure resources created for the cluster.
+                          type: object
+                          required:
+                            - key
+                            - value
+                          properties:
+                            key:
+                              description: key is the key of the tag
+                              type: string
+                              maxLength: 128
+                              minLength: 1
+                              pattern: ^[a-zA-Z][0-9A-Za-z_.=+-@]+$
+                            value:
+                              description: value is the value of the tag.
+                              type: string
+                              maxLength: 256
+                              minLength: 1
+                              pattern: ^[0-9A-Za-z_.=+-@]+$
+```
+
+### Implementation Details/Notes/Constraints [optional]
+Add a new field `resourceTags` to `.status.platformStatus.azure` of the 
+`infrastructure.config.openshift.io` type. Tags included in the `resourceTags` field 
+will be applied to new resources created for the cluster by the in-cluster operators.
+
+The `resourceTags` field in `status.platformStatus.azure` of `infrastructure.config.openshift.io`
+will be populated by the installer using the entries from `platform.azure.userTags` field of `install-config`.
+
+`status.platformStatus.azure` of `infrastructure.config.openshift.io` is immutable.
+
+All operators that create Azure resources will apply these tags to all Azure 
+resources they create. List of in-cluster operators managing cloud resources
+could vary across platform types, example for AWS there are additional operators
+like aws-ebs-csi-driver-operator, aws-efs-csi-driver-operator to manage specific
+resources.
+
+| Operator | Resources created by the operator |
+| -------- | ----------------------------- |
+| cluster-image-registry-operator | Storage Account |
+| cluster-ingress-operator | Load Balancer, DNS records |
```

Comment:
Thank you for confirming. Removed Load Balancer from the list of resources created by ingress operator.

Reply:
I had forgotten that we use a unified load balancer on Azure, which means that the ingress operator uses the existing LB.  However, if the ingress operator creates a service with `spec.type: LoadBalancer`, is there a resource for the public IP address or vnet that then needs to be tagged?  (Sorry, I'm a ignorant of the details of how LBs are configured on Azure.)

Reply:
For the record, Bharath and I discussed this, and no operator per se creates these resources; rather, the resources are created by the in-tree cloud provider implementation, which is out of scope for this EP.  

@bharath-b-rh, as you pointed out to me, the RFE excludes resources created by the in-tree cloud provider implementation.  It might be worth listing that as an explicit non-goal in the EP.

Reply:
I have included this in the drawbacks section [here](https://github.com/openshift/enhancements/pull/1191/files#diff-07e0630cc477ba74bc69ea6a15ecefd4f5ce32927210e4e7b0b89d423d2193dbR252-R254)